Add trip gui

BasharDimaria opened the pull request:
i combined Noora(With CSS ) task and karam (without CSS)  to my task and solved conflicts with johan and master brunch and test it , so everything is working .

ITJohan commented:
Nice! Har 3 grejer bara:

- Det verkar som att de gamla testerna inte går igenom, de måste ändras om det gjorts breaking changes
- Det måste skrivas tester till alla nya komponenter, antingen kollar ni på hur jag gjort eller så finns det en bra guide på https://www.robinwieruch.de/react-testing-library som jag följde
- Vet inte om ni såg vad vi skrev på Teams om Styled components, men vi tänkte köra det istället för en separat css fil eftersom det blir lättare att samarbeta i kodbasen. Guide finns på https://styled-components.com/docs/basics

AndreasWJ commented:
För att lägga till lite till Johans feedback:
Kom ihåg att alltid linta, det gör du med `npm lint` när du är i frontend mappen.
Kom även ihåg att ta alla nya uppdateringar från master, det kanske du redan gjort, påminner bara:
```
git checkout din-branch      # gets you "on branch din-branch"
git fetch origin        # gets you up to date with origin
git rebase master    # alternativt 'git merge origin/master'
```

BasharDimaria commented:
I have deleted Karam and Noora componenter becuse you want a lot of things such as test and lint and I can't dot it for 3 tasks so I deleted from my component and they said that they will do their tasks , :)

ITJohan commented on frontend/package.json:
```diff
@@ -3,12 +3,21 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^1.2.30",
+    "@fortawesome/free-brands-svg-icons": "^5.14.0",
+    "@fortawesome/free-regular-svg-icons": "^5.14.0",
+    "@fortawesome/free-solid-svg-icons": "^5.14.0",
+    "@fortawesome/react-fontawesome": "^0.1.11",
```
Eftersom Andreas har designat med Feather Icons (https://github.com/feathericons/react-feather), så snacka vi i Teams om att använda det istället. Du får gärna flika in med din egen mening. :relaxed:

ITJohan commented on frontend/src/App.js:
```diff
@@ -1,31 +1,35 @@
-    </Router>
-  )
-}
+class App extends Component {
```
Varför har du ändrat denna till en klasskomponent? Eller är det eslint som gjorde det? Jag föredrar funktionella komponenter eftersom man kan använda hooks då, se https://reactjs.org/docs/hooks-intro.html och ser gärna att denna ändras tillbaks.

BasharDimaria replied:
jag har inte gjort den , Eslint som gjorde tror jag , Eslont gör alldels för mycket onödiga error och det tar alldels alldels för mycket tid att fixa , inte bara hos mig utan också hos Karam och Noora , kan vi inte köra utan den ? om jag trycker eslint , den ordnar koden på sätt kan inte ens fattas.

AndreasWJ replied:
ESLint gör inget med komponent definitioner. Kan ha blivit så om ni ändrat mellan spacing->tab eller tvärtom i din IDE, kan även ha med LF och CRLF.

ESLint tycker jag vi fortsätter med. Det kommer inte bara hjälpa oss andra i gruppen att läsa kod, men kommer också hjälpa handledare om de någonsin kikar igenom vår kod. Dessutom är det standard i projektarbeten att komma överens över en särskild standard.

Jag kan dock starkt rekommendera https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
Installera den i din VS Code, den kommer effektivt kalla "npm lint" varje gång du gör en ändring och sedan visa hur du fixar problemen. Mycket mer effektivt att ändra koden medans man skriver den.

BasharDimaria replied:
Nu , körde jag Eslint , i mina filer , den visade ingen Fel på dem och då commited jag igen här , hoppas den ska vara bra :)

AndreasWJ replied:
Ser riktigt fint ut! Bra jobbat 👍

ITJohan commented on frontend/src/components/Addtrip.js:
```diff
@@ -0,0 +1,126 @@
+    </div>)
+}
+
+  render(){ 
+   return ( 
+    <div className="SearchGui">
+                <Formik data-testid="form"initialValues={{origin:"",destenation:"",seats:"1",date:"",time:""}} 
+                     onSubmit={this.onSubmit}
+                     render={this.form} />
+            </div>)
+ }
+  }
```
Samma problem med indentering här

ITJohan commented on frontend/src/style/Form.css:
```diff
@@ -0,0 +1,99 @@
+ .reg form{
```
Hela denna filen är onödig nu? I så fall kan den tas bort

ITJohan reviewed (changes requested):
Har en del grejer jag skulle villa se fixat innan vi mergar till master, du får gärna ge din egen åsikt om dem. :relaxed:

ITJohan commented on frontend/src/components/Addtrip.js:
```diff
@@ -0,0 +1,126 @@
+}
+
+form = (props) => {
+    return (
+    <div className="Addtrip">
+    <Form onSubmit={props.handleSubmit} className="Form">
+       <Icon icon={faMapMarkerAlt} className="icon" /> 
+       <Input  name="origin" placeholder="Origin" className="input"/>
+        <Icon icon={faMapMarkerAlt} className="icon" /> 
+        <Input name="destenation" placeholder="Destenation" className="input"/>
+        <div className="seats">
+          <Label>Seats</Label>
+          <br/>
+        <Input name="seats" component="select" className="seats">
+          <option value="1">1</option>
+          <option value="2">2</option>
+          <option value="3">3</option>
+          <option value="4">4</option>
+        </Input>
+        </div>
+       
+        <div >
+        <Label>Date</Label>
+        <br/>
+        <Input name="date" type="date" className="input">
+        </Input>
+        </div>
+        <div>
+        <Label>Time</Label>
+        <br/>
+        <Input name="time" type="time" className="input"></Input>
+        </div>
+        <i class="fal fa-bus-alt"></i>
+        <Button data-testid= "add-button" type="submit" className="Modal">ADD Ride</Button>
+        
+    </Form>
```
Här är det problem med indentering, kör gärna eslint i denna filen eller använd Prettier extension i VSCode, https://prettier.io/

ITJohan commented:
Skulle du kunna dra ned nya mastern till din branch? Det kommer uppstå lite merge conflicts men det är bara acceptera de från mastern och lägga till det du gjorde i App.js.

```
git checkout ADDTrip_GUI
git fetch
git merge origin/master
```

Du kan sedan köra `npm run lint-fix` för att fixa alla lint errors.

ITJohan commented:
@BasharDimaria Jag har en lokal version av ADDTrip_GUI där jag fixat alla problem och skulle kunna pusha den om du vill.

BasharDimaria commented:
jag löste precis Merge konflikter , det borde gå bra mu , jag har kört Lint i Mina filer (Addtrip.js och .test.js ) och allt är ok :)

ITJohan commented:
Linten går fortfarande inte igenom, du har addat fel version av .eslintrc.js. Tror även du har kört lint från root-mappen så du har lintat backend, så linta alltid från frontend mappen. Antingen får du reverta tillbaks de två commitsen lokalt och force-pusha nya commits till remote, eller så får du göra nya commits som fixar allt.

BasharDimaria commented:
jag öpnnar bara Frontend fil från början ,  linten visar ingen fel påmina filer , jag vet inte vad är felet som du pratar om ? alllt går som en klocka hos mig , jag mergat och allt funkar

BasharDimaria commented:
nu , tror jag att min task är färdig :)

ITJohan commented:
Nästan, som jag sa så har du den gamla versionen av eslintrc-filen. Har gjort en annan pull request som fixar alla problem nu, med dina commits intakta så att vi kan komma vidare, se #15.
